src/sparsezoo/v2/file.py

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+from typing import Any, Dict, Optional
+
+import onnx
+import yaml
+
+from PIL import Image
+from sparsezoo.utils.downloader import download_file
+from sparsezoo.utils.numpy import load_numpy_list
+
+
+__all__ = ["File"]
+
+
+class File:
+    """
+    Object to wrap around common files. Currently, supporting:
+    - numpy files
+    - onnx files
+    - markdown files
+    - json files
+    - csv files
+    - image files
+
+    :param name: name of the File
+    :param path: path of the File
+    :param url: url of the File
+    """
+
+    def __init__(
+        self, name: str, path: Optional[str] = None, url: Optional[str] = None
+    ):
+        self.name = name
+        self.url = url
+        self.path = path
+
+        # self.path can have any extension, including no extension.
+        # However, the File object also contains information
+        # About its loadable extensions.
+        # Loadable files can be read into the memory.
+
+        self.loadable_extensions = {
+            ".npz": self._validate_numpy,
+            ".onnx": self._validate_onnx,
+            ".md": self._validate_markdown,
+            ".json": self._validate_json,
+            ".csv": self._validate_csv,
+            ".jpg": self._validate_img,
+            ".png": self._validate_img,
+            ".jpeg": self._validate_img,
+            ".yaml": self._validate_yaml,
+        }
+
+    @classmethod
+    def from_dict(cls, file: Dict[str, Any]) -> "File":
+        """
+        Factory method for creating a File class object
+        from a file dictionary
+        (useful when working with the `request_json` from NeuralMagic).
+        :param file: a dictionary which contains an information about
+            the file (as returned by NeuralMagic API)
+        :return: File class object
+        """
+        name = file.get("display_name")
```

Comment:
Wouldn't this be a bit redundant? I mean this information is already propagated in the type hint of the method's argument right?

Reply:
I agree but types aren't enforced in Python. I guess a `ValueError` would be nicer and more informative for a user rather than the `AttributeError` which would get thrown when python can't find `get` attribute on the `file` object